akka-http-core/src/main/scala/akka/http/scaladsl/settings/ServerSentEventSettings.scala

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (C) 2009-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.http.scaladsl.settings
+
+import akka.annotation.{ ApiMayChange, DoNotInherit }
+import akka.http.impl.settings.ServerSentEventSettingsImpl
+import com.typesafe.config.Config
+
+/**
+ * Public API but not intended for subclassing
+ *
+ * Options that are in "preview" or "early access" mode.
+ * These options may change and/or be removed within patch releases
+ * without early notice (e.g. by moving them into a stable supported place).
+ */
+@ApiMayChange @DoNotInherit
+abstract class ServerSentEventSettings private[akka] () extends akka.http.javadsl.settings.ServerSentEventSettings {
+  self: ServerSentEventSettingsImpl ⇒
+
+  override def maxEventSize: Int
+  override def maxLineSize: Int
+
+  override def withMaxEventSize(newValue: Int): ServerSentEventSettingsImpl = self.copy(maxEventSize = newValue)
```

Comment:
Same here but with the scaladsl version

```suggestion
  override def withMaxEventSize(newValue: Int): ServerSentEventSettings = self.copy(maxEventSize = newValue)
```